src/app/shared/components/city-autocomplete/city-autocomplete.component.ts

```diff
@@ -1,84 +1,48 @@
-import { Component, EventEmitter, OnInit, Output, Input } from '@angular/core';
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
 import { FormControl } from '@angular/forms';
 import { Select, Store } from '@ngxs/store';
 import { Observable, Subject } from 'rxjs';
-import { debounceTime, distinctUntilChanged, map, startWith, takeUntil } from 'rxjs/operators';
+import { debounceTime, distinctUntilChanged, startWith, takeUntil } from 'rxjs/operators';
 import { MetaDataState } from '../../store/meta-data.state';
-import { CityList } from '../../store/meta-data.actions';
+import { ClearCities, GetCities } from '../../store/meta-data.actions';
 import { MatAutocompleteSelectedEvent } from '@angular/material/autocomplete';
 import { City } from '../../models/city.model';
-import { FormGroup } from '@angular/forms';
-import { CityFilterService } from '../../services/filters-services/city-filter/city-filter.service';
 
 @Component({
   selector: 'app-city-autocomplete',
   templateUrl: './city-autocomplete.component.html',
   styleUrls: ['./city-autocomplete.component.scss']
 })
 export class CityAutocompleteComponent implements OnInit {
+
   @Output() selectedCity = new EventEmitter();
-  @Input() address: FormGroup;
 
-  city: City;
   cityControl = new FormControl();
   cities: City[] = [];
-  noCity = false;
   destroy$: Subject<boolean> = new Subject<boolean>();
 
-  @Select(MetaDataState.filteredCities)
-  filteredCities$: Observable<City[]>;
-
+  @Select(MetaDataState.cities)
+  cities$: Observable<City[]>;
+  @Select(MetaDataState.isCity)
+  isCity$: Observable<boolean[]>;
 
-  constructor(public filterCityService: CityFilterService, public store: Store) {
-  }
+  constructor(public store: Store) { }
 
   ngOnInit(): void {
-    this.filterCityService.fetchCities()
-      .subscribe((data) => {
-        this.cities = data;
-      });
+    this.cities$
+      .pipe(
+        takeUntil(this.destroy$),
+      ).subscribe((cities) => this.cities = cities);
 
     this.cityControl.valueChanges
       .pipe(
         takeUntil(this.destroy$),
         debounceTime(300),
         distinctUntilChanged(),
         startWith(''),
-      ).subscribe(value => {
-      if (value) {
-        this.store.dispatch(new CityList(this._filter(value)));
-      } else {
-        this.store.dispatch(new CityList([]));
-      }
-      ;
-    });
-    this.address && this.address.valueChanges.subscribe(address => this.cityControl.setValue(address.city));
+      ).subscribe(value =>
+        (value.length > 2) ? this.store.dispatch(new GetCities(value)) : this.store.dispatch(new ClearCities()));
```

Comment:
move check to filter operator;
why do we need to clear cities?

Reply:
@dmitryy90  because when I for example tapped something unexisting, then I need to display no such city. I push this thing to state. But then if I clear the input again and won't make new GetCities yet, I will still see "No such City", because we have it from teh state

Reply:
and we do not have the filtration on the front-end, it's on the back-end. 
If I tap 3-5 characters I will have no more than 8-12 cities, usually, it's 8. SO I don't see the point to make the filtration again.

Reply:
that's why I have deleted filtration at all. Cause I thought we don't need to make filtration for 12 cities again.
Should I revet it?

Reply:
I mean put length check in pipe(filter()) for avoiding triggering of subscription

Reply:
lets talk on tomorrow's meeting about clear cities action